BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -145,6 +174,10 @@ extension NetworkStore: BraveWalletJsonRpcServiceObserver {
     updateChainList()
   }
   public func chainChangedEvent(_ chainId: String, coin: BraveWallet.CoinType) {
-    setSelectedChain(chainId: chainId, coin: coin)
+    walletService.setSelectedCoin(coin)
+    Task { @MainActor in
+      guard let chain = allChains.first(where: { $0.chainId == chainId && $0.coin == coin }) else {  return }
+      await setSelectedChain(chain)
+    }
   }
 }
```

Comment:
shouldn't this store listening to `accountsChanged` or maybe `keyringCreated` and auto setSelectedAccount with that newly created account

Reply:
Fixed in [d60fbfc](https://github.com/brave/brave-ios/pull/5681/commits/d60fbfcebc7894a76dccaef08d7278f03f36f428)